moduleframework/common.py

```diff
@@ -24,23 +25,46 @@
 Custom configuration and debugging library.
 """
 
-from __future__ import print_function
-
 import netifaces
 import socket
 import os
 import urllib
 import yaml
 import subprocess
 import copy
-import sys
 import random
 import string
 import requests
 import warnings
 import ast
-from avocado.utils import process
-from moduleframework.mtfexceptions import ModuleFrameworkException, ConfigExc, CmdExc, DefaultConfigExc
+import glob
+import avocado.utils
+import mtfexceptions
+import core
+
+
+class MTFConfParser(dict):
```

Comment:
@jscotka What about something like this:

~~~
class Config(dict):
     def __getattr__(self, key):
         try:
             return self[key]
         except KeyError:
             raise AttributeError
     def __setattr__(self, key, value):
         self[key] = value
     def __init__(self, yaml_file, release="current"):
         config = yaml.load(yaml_file)
         self["raw"] = config
         pdc = config["pdc"]
....
~~~

And you have access to all fields.

Reply:
@phracek  does that have  some benefits instead of actual implementation to redefine these getters and setters?
I trough to redefine ``__getattr__`` to lead some easier to debug exceptions (like, show path to file where is this found, and show value from default config, what should not be modifed), but it could be further improvement

Reply:
yeah let's leave it for now.